File: banking-component/src/test/java/de/rwth/swc/banking/BankingApi.java
package de.rwth.swc.banking;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.service.annotation.PostExchange;

public interface BankingApi {

    @PostExchange("/v1/transfer")
    ResponseEntity<Boolean> sendMoneyV1(@RequestBody Transfer transfer);

    @PostExchange("/v2/transfer")
    ResponseEntity<Boolean> sendMoneyV2(@RequestBody Transfer transfer);
}

File: banking-component/src/test/kotlin/de/rwth/swc/banking/BankingApi.kt
package de.rwth.swc.banking

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.service.annotation.PostExchange

interface BankingApi {
    @PostExchange("/api/v1/transfer")
    fun transferV1(@RequestBody transfer: Transfer): ResponseEntity<TransferResponse>

    @PostExchange("/api/v2/transfer")
    fun transferV2(@RequestBody transfer: Transfer): ResponseEntity<TransferResponse>
}